Source/core/CyclicBuffer.h

```diff
@@ -94,6 +96,8 @@ namespace Core {
 
             uint32_t GetCompleteTail(uint32_t offset) const
             {
+                ASSERT(_Parent._administration->_tailIndexMask < std::numeric_limits<uint32_t>::max());
```

Comment:
Maybe let's use static cast in order to get a maximum, since then we do not need to include `limits`.
So something like this instead:
```
static_cast<uint32_t>(~0)
```

Reply:
Agreed.